internal/factsengine/gatherers/dir_scan_test.go

```diff
@@ -0,0 +1,207 @@
+package gatherers_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+	"github.com/trento-project/agent/internal/factsengine/gatherers"
+	"github.com/trento-project/agent/internal/factsengine/gatherers/mocks"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+)
+
+const dirScanTestBasePath = "/var/test"
+const dirScanTestGlobPattern = "/var/test/*/ASCS*"
+
+type DirScanGathererSuite struct {
+	suite.Suite
+	testFS     afero.Fs
+	basePathFS string
+}
+
+func TestDirScanGathererSuite(t *testing.T) {
+	suite.Run(t, new(DirScanGathererSuite))
+}
+
+func (s *DirScanGathererSuite) SetupSuite() {
+	bfs := afero.NewOsFs()
+
+	s.basePathFS = afero.GetTempDir(bfs, "")
+	tFs := afero.NewBasePathFs(bfs, s.basePathFS)
+	for i := 0; i <= 2; i++ {
+		dirPath := fmt.Sprintf("%s/%d/", dirScanTestBasePath, i)
```

Comment:
`filepath.Join` looks more appropriate to create the file paths, instead of `Sprintf`

Reply:
you are right but due to the nature of paths, interpolations, convertion to string from int for the index, it's pretty deterministic and it's a test so I used sprintf on purpose to avoid other boilerplate code